subsystem/day2_cluster_test.go

```diff
@@ -0,0 +1,326 @@
+package subsystem
+
+import (
+	//"archive/tar"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	//"io"
+	//"io/ioutil"
+	//"os"
+	//"os/exec"
+	//"reflect"
+	"time"
+
+	//"github.com/openshift/assisted-service/internal/bminventory"
+	//"github.com/openshift/assisted-service/internal/host"
+
+	//"github.com/alecthomas/units"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
+
+	//"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/models"
+)
+
+const (
+	statusInfoAddingHosts = "cluster is adding hosts to existing OCP cluster"
+)
+
+var _ = Describe("Day2 cluster tests", func() {
+	ctx := context.Background()
+	var cluster *installer.RegisterAddHostsClusterCreated
+	var clusterID strfmt.UUID
+	var err error
+	AfterEach(func() {
+		clearDB()
+	})
+
+	BeforeEach(func() {
+
+		cluster, err = userBMClient.Installer.RegisterAddHostsCluster(ctx, &installer.RegisterAddHostsClusterParams{
+			NewAddHostsClusterParams: &models.AddHostsClusterCreateParams{
+				Name:             swag.String("test-cluster"),
+				OpenshiftVersion: swag.String("4.6"),
+				APIVipDnsname:    swag.String("api_vip_dnsname"),
+				ID:               strToUUID(uuid.New().String()),
+			},
+		})
+		By(fmt.Sprintf("clusterID is %s", *cluster.GetPayload().ID))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(swag.StringValue(cluster.GetPayload().Status)).Should(Equal("adding-hosts"))
+		Expect(swag.StringValue(cluster.GetPayload().StatusInfo)).Should(Equal(statusInfoAddingHosts))
+		Expect(cluster.GetPayload().StatusUpdatedAt).ShouldNot(Equal(strfmt.DateTime(time.Time{})))
+	})
+
+	JustBeforeEach(func() {
+		clusterID = *cluster.GetPayload().ID
+	})
+
+	generateApiVipPostStepReply := func(h *models.Host, success bool) {
+		checkVipApiResponse := models.APIVipConnectivityResponse{
+			IsSuccess: success,
+		}
+		bytes, jsonErr := json.Marshal(checkVipApiResponse)
+		Expect(jsonErr).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				StepType: models.StepTypeAPIVipConnectivityCheck,
+				Output:   string(bytes),
+				StepID:   "apivip-connectivity-check-step",
+			},
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+
+	It("cluster CRUD", func() {
+		_ = registerHost(clusterID)
+		Expect(err).NotTo(HaveOccurred())
+		getReply, err1 := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+		Expect(err1).NotTo(HaveOccurred())
+		Expect(getReply.GetPayload().Hosts[0].ClusterID.String()).Should(Equal(clusterID.String()))
+
+		getReply, err = agentBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
```

Comment:
is checking the agent client relevant for this flow?

Reply:
yes, just to see that agent can access cluster